dataprofiler/data_readers/data_utils.py

```diff
@@ -564,3 +564,21 @@ def load_as_str_from_file(file_path, file_encoding, max_lines=10,
                 break
             
     return data_as_str
+
+def is_stream_buffer(filepath_or_buffer):
+    """
+    Determines whether a given argument is a filepath or buffer.
+    
+    :param filepath_or_buffer: path to the file or buffer 
+    :type filepath_or_buffer: str
+    :param encoding: File encoding
+    :type encoding: str
+    :param seek: position to start in buffer
+    :type seek: int
+    :return: true if string is a buffer or false if string is a filepath
+    :rtype: boolean
+    """
+
+    if isinstance(filepath_or_buffer, (StringIO, BytesIO)):
+        return True
+    return False
```

Comment:
I think once we added the context manager we can just incorporate the simple true / false in the class below as described in other comments